packages/cli-kit/src/public/common/string.ts

```diff
@@ -113,3 +114,87 @@ export function getRandomName(family: RandomNameFamily = 'business'): string {
   }
   return `${takeRandomFromArray(mapping[family].adjectives)}-${takeRandomFromArray(mapping[family].nouns)}`
 }
+
+/**
+ * Given a string, it returns it with the first letter capitalized
+ */
+export function capitalize(str: string): string {
+  return str.substring(0, 1).toUpperCase() + str.substring(1)
+}
+
+/**
+ * Try to convert a string to an int, falling back to undefined if unable to
+ */
+export function tryParseInt(maybeInt: string | undefined): number | undefined {
+  let asInt: number | undefined
+  if (maybeInt !== undefined) {
+    asInt = parseInt(maybeInt, 10)
+    if (isNaN(asInt)) {
+      asInt = undefined
+    }
+  }
+  return asInt
+}
+
+/**
+ * Given a series of rows inside an array, where each row is an array of strings (representing columns)
+ * Parse it into a single string with the columns aligned
+ */
+export function linesToColumns(lines: string[][]): string {
```

Comment:
What about having a module, `@shopify/cli-kit/node/terminal`, that exports this function as `buildTableString`.